modules/kameleonvolume/tasks/kameleonvolumetorawtask.cpp

```diff
@@ -0,0 +1,207 @@
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this  *
+ * software and associated documentation files (the "Software"), to deal in the Software *
+ * without restriction, including without limitation the rights to use, copy, modify,    *
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to    *
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/kameleonvolume/tasks/kameleonvolumetorawtask.h>
+#include <modules/kameleonvolume/kameleonvolumereader.h>
+#include <modules/volume/rawvolume.h>
+#include <modules/volume/rawvolumewriter.h>
+
+#include <openspace/documentation/verifier.h>
+
+#include <ghoul/misc/dictionaryjsonformatter.h>
+#include <ghoul/filesystem/filesystem.h>
+
+#include <ghoul/logging/logmanager.h>
+#include <ghoul/misc/dictionaryluaformatter.h>
+
+#include <fstream>
+
+namespace {
+    const char* KeyInput = "Input";
+    const char* KeyRawVolumeOutput = "RawVolumeOutput";
+    const char* KeyDictionaryOutput = "DictionaryOutput";
+    const char* KeyDimensions = "Dimensions";
+    const char* KeyVariable = "Variable";
+    const char* KeyTime = "Time";
+    const char* KeyLowerDomainBound = "LowerDomainBound";
+    const char* KeyUpperDomainBound = "UpperDomainBound";
+
+    const char* KeyMinValue = "MinValue";
+    const char* KeyMaxValue = "MaxValue";
+
+    const char* _loggerCat = "KameleonVolumeToRawTask";
+}
+
+namespace openspace {
+namespace kameleonvolume {
+
+KameleonVolumeToRawTask::KameleonVolumeToRawTask(
+    const ghoul::Dictionary& dictionary)
+    : _autoDomainBounds(false)
+{
+    try {
+        openspace::documentation::testSpecificationAndThrow(
+            documentation(),
+            dictionary,
+            "KameleonVolumeToRawTask"
+        );
+    } catch (const ghoul::RuntimeError& e) {
```

Comment:
It's probably better not to catch this and let the caller deal with the exception.  If you don't want to, there is a `testSpecification` method that has a return value